### PR TITLE
Add a conditional block container to the documentation. #9309

### PR DESCRIPTION
**Pull request by jansiegel**
### Context
This PR adds a container used to display/hide blocks of content relevant to specific frameworks to Vuepress.

It relies on setting up an environmental variable of `FRAMEWORK` to the desired framework, for example: `FRAMEWORK=react`.

[skip changelog]

### How has this been tested?
Tested manually by adding a block of 
```
::: only-for react
Content for React
:::
```

and setting up the `FRAMEWORK` environmental variable.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9309 

### Affected project(s):
- [ ] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

**Comment by jansiegel** [comment]
@wszymanski I replaced the frontmatter-based framework-retrieving method with a helper copied from your PR (#9371), which takes it from the `FRAMEWORK` environmental variable. Could you re-review and see if it makes sense?

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+/* eslint-disable jsdoc/require-description-complete-sentence */
+/**
+ * Container used to display/hide blocks of content relevant to specific frameworks.
+ * It relies on providing the framework name(s) as arguments to the container (see 'Usage' below) and setting a
+ * `framework` field in front matter as the framework name, for example:
+ *
+ * ```
+ * ---
+ * framework: vue
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:11]: Is it still valid? I don't understand which part of code handle it.

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ *
+ * ::: only-for js react vue
+ * Content to be displayed only for JS, React and Vue documentation.
+ * :::
+ * ```
+ */
+module.exports = {
+  type: 'only-for',
+  render(tokens, index) {
+    const framework = getBuildDocsFramework() || 'js';
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:29]: I'm using `javascript` phrase for default framework: https://github.com/handsontable/handsontable/blob/3e27d756efa7a8d774859c9827cdf0fb2acfa2c1/docs/.vuepress/helpers.js#L73-L75.

Should we sync it?

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js (hunk 1 above)
Comment (jansiegel) [reply docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:11]: You're right, not applicable anymore 👍

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js (hunk 2 above)
Comment (jansiegel) [reply docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:29]: I'll change it to `javascript` for now, but let's change it to `getDefaultFramework()` after merging both of the PRs to the epic branch.

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * ::: only-for javascript react vue
+ * Content to be displayed only for JS, React and Vue documentation.
+ * :::
+ * ```
+ */
+module.exports = {
+  type: 'only-for',
+  render(tokens, index) {
+    const framework = getBuildDocsFramework() || 'javascript';
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:27]: The `getBuildDocsFramework` won't work for `npm run docs:start` script, so sorry for this suggestion. There should be used something else. Maybe `parseFramework` helper will be okey.

https://github.com/handsontable/handsontable/blob/3e77e0bc759b2c2aaa20b2a3ef68daaac13ef51f/docs/.vuepress/helpers.js#L156-L162

Please change it.

**Review by wszymanski — CHANGES_REQUESTED** [review]

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * ::: only-for javascript react vue
+ * Content to be displayed only for JS, React and Vue documentation.
+ * :::
+ * ```
+ */
+module.exports = {
+  type: 'only-for',
+  render(tokens, index, opts, env) {
+    const framework = getContainerFramework(env.relativePath) || getDefaultFramework();
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:30]: Works good with webpack config ❤️

File: docs/.vuepress/containers/helpers/index.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  const frontMatterEntriesCount = Object.keys(frontMatterObj).reduce(
+    (sum, key) => {
+      if (Array.isArray(frontMatterObj[key])) {
+        return sum + frontMatterObj[key].length + 1;
+      }
+
+      return sum + 1;
+    }, 0);
+
+  // Adding `3` to compensate for the frontmatter syntax.
```
Comment (wszymanski) [inline docs/.vuepress/containers/helpers/index.js:23]: Please move to variable and describe it within comment.

File: docs/.vuepress/containers/helpers/index.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+ * @returns {number} The number of lines taken by the frontmatter.
+ */
+function getContainerFrontMatterLength(frontMatterObj) {
+  const frontMatterEntriesCount = Object.keys(frontMatterObj).reduce(
+    (sum, key) => {
+      if (Array.isArray(frontMatterObj[key])) {
+        return sum + frontMatterObj[key].length + 1;
+      }
+
+      return sum + 1;
```
Comment (wszymanski) [inline docs/.vuepress/containers/helpers/index.js:20]: Please use more descriptive name for the `sum` variable. Maybe `nrOfLines` will be better.

```suggestion
    (nrOfLines, key) => {
      if (Array.isArray(frontMatterObj[key])) {
        return nrOfLines + frontMatterObj[key].length + 1;
      }

      return nrOfLines + 1;
```

**Review by wszymanski — CHANGES_REQUESTED** [review]
It looks good. Please just check my suggestions and update `helpers.js` file.

**Comment by jansiegel** [comment]
> It looks good. Please just check my suggestions and update `helpers.js` file.

Same as in #9384:

I think it'll make more sense to sync it after (or during) the merge. I copied the helper.js file to emulate the API available in https://github.com/handsontable/handsontable/issues/9307, but if there are no significant changes in the API naming I'd leave it as is. It'll have to be tested and possibly corrected on the epic branch when the actual build logic will be implemented anyway.

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+const {
+  getDefaultFramework
+} = require('../../helpers');
+const { getContainerFramework } = require('../helpers');
+
+/* eslint-disable jsdoc/require-description-complete-sentence */
+/**
+ * Container used to display/hide blocks of content relevant to specific frameworks.
+ * It relies on providing the framework name(s) as arguments to the container (see 'Usage' below) and setting a
+ * `FRAMEWORK` environmental variable as the framework name, for example:
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:10]:
```suggestion
 * `DOCS_FRAMEWORK` environmental variable as the framework name, for example:
```

File: docs/.vuepress/containers/conditionalBlock/conditionalBlock.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+ *
+ * ```
+ * FRAMEWORK=react
```
Comment (wszymanski) [inline docs/.vuepress/containers/conditionalBlock/conditionalBlock.js:13]:
```suggestion
 * DOCS_FRAMEWORK=react
```

**Review by wszymanski — COMMENTED** [review]
Looks good, just missed two changes... sorry.

File: docs/.vuepress/tools/build.mjs [hunk 9]
```diff
@@ -7,6 +7,9 @@ import { getVersions } from '../helpers.js';
 const { logger, spawnProcess } = utils;
 const __dirname = path.dirname(fileURLToPath(import.meta.url));
 const buildMode = process.env.BUILD_MODE;
+const [ ...cliArgs ] = process.argv.slice(2);
+
+const NO_CACHE = cliArgs.some(opt => opt.indexOf('--no-cache') !== -1);
```
Comment (wszymanski) [inline docs/.vuepress/tools/build.mjs:12]:
```suggestion
const NO_CACHE = cliArgs.some(opt => opt.includes('--no-cache'));
```

**Review by wszymanski — APPROVED** [review]

**Pull request merged**